fastlane: slightly improve formatting for full descriptions

IzzySoft opened the pull request:
This PR slightly improves formatting of the English full description, making it possible to render it as Markdown (supported e.g. at IzzyOnDroid) while keeping it compatible with places not supporting it (e.g. F-Droid.org, PlayStore).

Please forgive me not having applied it to all locales, but only to those with character sets I knew how to deal with :see_no_evil:

IzzySoft committed:
fastlane: slightly improve formatting for full descriptions

IzzySoft commented:
@Razeeman any chance to get this merged? Would be quite good for [the app's listing at IzzyOnDroid](https://apt.izzysoft.de/packages/com.razeeman.util.simpletimetracker). And in case you're curious, some stats:

<img width="1247" height="415" alt="image" src="https://github.com/user-attachments/assets/2076e2c4-dc40-4027-b6ed-1cc829d5e7bd" />

Razeeman commented:
Hello! Can you please share how does it look before the changes? Does it not render currently? 

Description is very important for Google Play, so I'd rather not change it without running A/B tests.

IzzySoft commented:
> Can you please share how does it look before the changes? Does it not render currently?

To do that, I'd need to void the changes here, create an index with that, and take screenshots with multiple clients (look might depend on that), then revert the changes, and rebuild the index again – which would take more than an hour I currently don't have… But looking at the diff, let me simulate it here:

----
* <b>Simple interface</b>App has a minimalistic interface that is very easy to use.

* <b>Widgets</b>Track your activities directly from your home screen.

* <b>Works offline and respects your privacy</b>App doesn't require an Internet connection or account registration. Your data never leaves your phone. Neither the developers nor any third-parties have access to it.

----

versus

----
* <b>Simple interface:</b> App has a minimalistic interface that is very easy to use.
* <b>Widgets:</b> Track your activities directly from your home screen.
* <b>Works offline and respects your privacy:</b> App doesn't require an Internet connection or account registration. Your data never leaves your phone. Neither the developers nor any third-parties have access to it.

-----

is the closest I can get here. In short: each bullet point would become its own list, and not all line breaks would be recognized with "unflavored" standard markdown.

> Description is very important for Google Play, so I'd rather not change it without running A/B tests.

Sorry, I cannot run those. It's plain text for Google, and nothing much changes there. If you insist, the graphical bullet points could be restored (we meanwhile updated our importer so it now recognizes and converts them on import). The most important thing there is the empty lines between the bullet points, which would break the list here, and the line break inside each bullet point which would get lost.

Razeeman commented:
Thank you for the explanation. If I understood correctly, there is a parser that parses bullet point symbol as a bulleted list, and formatting gets broken. Is there a possibility to disable this bulleted list parsing and show description as simple text like in Google Play? Maybe only for this repository? Or adjust it to not remove new lines?

It seems somewhat limiting to change description to  get a correct bulleted list working.

IzzySoft commented:
It is possible to show it as "plain text", and that can done per-app, yes. It just doesn't look half as good then. So I'd not count it "limiting", but rather "improving". This is how it looks with the improvements:

<img width="929" height="490" alt="image" src="https://github.com/user-attachments/assets/3556b1a5-5460-4386-ae11-db720fd60ce1" />

and this is how it looks without:

<img width="928" height="566" alt="image" src="https://github.com/user-attachments/assets/16212d1b-e304-4195-804e-69180954e9af" />

Bullet points are not properly aligned, and there is no "text flow" – as each line break in the full description is simply replaced with a `<br>`. Do you really prefer that?

Razeeman commented:
It seems more like a "to each is own" stylistic choice. For me, text on the second screenshot seems easier to read, it is separated, has more space. Hope this makes sense.

IzzySoft commented:
Not forcing this upon you – it's your app, so your choice. If you want us (IzzyOnDroid) to switch to F-Droid like plain-text interpretation, we can of course do that – and close this PR unmerged (with the proper cleanup).

IzzySoft commented:
So for clarity's sake: shall I withdraw my PR then, and switch interpretation to "plain text" (i.e. just replacing line breaks with `<br>`, as in the second screenshot)? No bad feelings, I just need to know how we shall proceed here.

Razeeman commented:
Sorry, got carried away. Yes, this is probably would be best for now. Thank you for the understanding!

IzzySoft commented:
All good! It's your app, so your decision. No bad feelings at all! Cleaned up on our end then, and switched the config to pick up your variant, converting line breaks to `<br>` as discussed. Thanks for considering – and your decision is fully respected :wink: Cleaning up here then.
